src/azure-cli/azure/cli/command_modules/backup/custom_help.py

```diff
@@ -671,11 +672,25 @@ def validate_and_extract_container_type(container_name, backup_management_type):
     return None
 
 
-def validate_update_policy_request(existing_policy, new_policy):
+def validate_update_policy_request(existing_policy, new_policy, yes=False):
     existing_backup_management_type = existing_policy.properties.backup_management_type
     new_backup_management_type = new_policy.properties.backup_management_type
     if existing_backup_management_type != new_backup_management_type:
         raise CLIError("BackupManagementType cannot be different than the existing type.")
+    # vault -> snapshot
+    if hasattr(existing_policy.properties, 'vault_retention_policy') and existing_policy.properties.vault_retention_policy is not None and hasattr(new_policy.properties, 'retention_policy') and new_policy.properties.retention_policy is not None:
+        raise CLIError(
+            """
+            Switching the backup tier from vaulted backup to snapshot is not possible. 
+            Please create a new policy for snapshot-only backups.
+            """)
+    # snapshot -> vault
+    if not yes and hasattr(existing_policy.properties, 'retention_policy') and existing_policy.properties.retention_policy is not None and hasattr(new_policy.properties, 'vault_retention_policy') and new_policy.properties.vault_retention_policy is not None:
+        warning_prompt = ('Changing the backup tier keeps current snapshots as-is under the existing policy. Future backups will be stored in the vault with new retention settings.' 
+                          'This action is irreversible and incurs additional costs. Switching from vault to snapshot requires reconfiguration.' 
+                          'Learn more at https://learn.microsoft.com/en-us/azure/backup/azure-file-share-backup-overview?tabs=snapshot.')
```

Comment:
Instruct the user here. Right now it's just a statement asking for a y/n input, and if they press "no" then the operation gets cancelled. Add a line that says "do you want to continue"